isis/src/docsys/documents/DeveloperAddFeature/DeveloperAddFeature.xml

```diff
@@ -0,0 +1,187 @@
+            <ol>
+              <li>Setup a GitHub account</li>
+              <li>Fork the ISIS repository</li>
+              <li>Clone your fork to a local repository</li>
+            </ol>
+
+          <h2><a name="Setting up an environment to compile and link ISIS">Setting up an environment to compile and link ISIS</a></h2>
+            <p>
+              The ISIS package is written in C++. In order to run and test your new feature, you will need to compile the source code and
+              link it against the ISIS library and the external libraries. Follow the instructions on the ISIS GitHub Wiki for 
+              <a href="https://github.com/USGS-Astrogeology/ISIS3/wiki/Developing-ISIS3-with-cmake"> building ISIS</a>.
+            </p>
+            <p>
+              Depending on how your computer is setup, you may also need to install:
+            </p>
+            <ul>
+              <li> Compiler (Mac OS Xcode, Linux GNU C++)</li>
+              <li> Editor (Vim, Nano, Atom, ...</li>
+            </ul>
+
+            <p>
+              Once you have a complete environment set up for ISIS development, we suggest you 
+              <a href="https://github.com/USGS-Astrogeology/ISIS3/wiki/Developing-ISIS3-with-cmake#building-isis3">compile and build</a> 
+              all of ISIS. This step will make sure sure you have everything setup and ready to go.
+            </p>
+
+          <h2><a name="Starting a conversation with other ISIS developers">Starting a conversation with other ISIS developers</a></h2>
+            <p>
+              Before you start writing your new feature, create an <a href="https://github.com/USGS-Astrogeology/ISIS3">issue</a> for 
+              it on the ISIS GitHub site. Doing this will allow other ISIS developers to way in on your preposed feature. By posting
+              a description of what you want to do and asking questions, other developers can give you pointers on how to 
+              implement it.
+            </p>
+
+          <h2><a name="Writing your new application">Writing your new application</a></h2>
+
+            <h3>Creating the files for your new application</h3>
+            <p>
+              The ISIS source code is organized by having a dedicated directory where all of the source code for an application is
+              stored. The files in this directory will be exclusively to your feature. This is where using another application as
+              a template can be very useful.
+            </p>
+            <ol>
+                <li>Decide where your new application belongs</li>
+                <li>Create a directory using the same name as you new application</li>
+                <li>Create a "main.cpp" file</li>
+                <li>Create a "your_new_app.cpp" file</li>
+                <li>Create a "your_new_app.xml" file</li>
+                <li>Create any additional support files (e.g., support_class.cpp, support_class.h</li>
```

Comment:
Change these filenames to be all lowercase for app and upper camel case for classes to match code style.